transport/wseb/src/main/java/org/kaazing/gateway/transport/wseb/WsebAcceptor.java

```diff
@@ -877,23 +877,29 @@ private IoHandler selectDownstreamHandler(ResourceAddress address,
         @Override
         protected void doExceptionCaught(HttpAcceptSession session, Throwable cause) throws Exception {
             // HTTP exception occurred, usually due to underlying connection failure during HTTP response
-            WsebSession wseSession = SESSION_KEY.get(session);
-            if (wseSession != null && !wseSession.isClosing()) {
-                wseSession.reset(cause);
-            }
-            else {
-                if (logger.isDebugEnabled()) {
-                    String message = format("Exception while handling HttpSession to create WsebSession: %s", cause);
-                    if (logger.isTraceEnabled()) {
-                        // note: still debug level, but with extra detail about the exception
-                        logger.debug(message, cause);
-                    }
-                    else {
-                        logger.debug(message);
-                    }
+
+            if (logger.isDebugEnabled()) {
+                String message = format("Exception while handling HttpSession to create WsebSession: %s", cause);
+                if (logger.isTraceEnabled()) {
+                    // note: still debug level, but with extra detail about the exception
+                    logger.debug(message, cause);
                 }
-                session.close(true);
+                else {
+                    logger.debug(message);
+                }
+            }
+            session.close(true);
+        }
+
+        @Override
+        protected void doSessionClosed(HttpAcceptSession session) throws Exception {
+            WsebSession wsebSession = SESSION_KEY.remove(session);
+            if (wsebSession != null && !wsebSession.isClosing()) {
+                wsebSession.reset(new Exception("Network connectivity has been lost or transport was closed at other end").fillInStackTrace());
             }
+
+            IoFilterChain filterChain = session.getFilterChain();
+            removeBridgeFilters(filterChain);
```

Comment:
Do we need to call wsebSession.getProcessor().remove(wsebSession)? Just asking because that's what you are doing in some of the other acceptors.